salt/utils/dns.py

```diff
@@ -1216,5 +1204,67 @@ def parse_resolv(src="/etc/resolv.conf"):
             "search": search,
             "options": options,
         }
-    except IOError:
+    except OSError:
+        return {}
+
+
+def parse_systemd_resolve():
+    """
+    Parse the information from systemd-resolve
+    """
+
+    nameservers = []
+    ip4_nameservers = []
+    ip6_nameservers = []
+    domain = ""
+
+    cmd = "systemd-resolve --status"
+    cmd = __salt__["cmd.run_all"](
+        "{}".format(cmd), python_shell=False, output_loglevel="quiet"
+    )
+
+    if cmd["retcode"] == 0:
+        prev_section = None
+        lines = [re.sub("[\t ]", "", line) for line in cmd["stdout"].splitlines()]
+
+        contents = {}
+        prev_section = None
+        for line in lines:
+            section = None
+            if ":" in line and line.count(":") <= 1:
```

Comment:
If `:` is in line, then there will always be one, so, `and line.count(":") == 1`?

Reply:
Yup.  Already made that change, just need to commit and push.